services/api/service.go

```diff
@@ -635,6 +750,38 @@ func (api *RelayAPI) updateProposerDuties(headSlot uint64) {
 	api.log.Infof("proposer duties updated: %s", strings.Join(_duties, ", "))
 }
 
+func (api *RelayAPI) updateOptimisticSlot(headSlot uint64) {
+	// Wait until there are no optimistic blocks being processed. Then we can
+	// safely update the slot.
+	api.optimisticBlocks.Wait()
+	api.optimisticSlot.Store(headSlot + 1)
+
+	builders, err := api.db.GetBlockBuilders()
+	if err != nil {
+		api.log.WithError(err).Error("unable to read block builders from db, not updating builder cache")
+		return
+	}
+	for _, v := range builders {
+		collStr := v.Collateral
+
+		// Try to parse builder collateral string to big int.
+		var builderCollateral big.Int
+		err = builderCollateral.UnmarshalText([]byte(collStr))
+		if err != nil {
+			api.log.WithError(err).Error("could not parse builder collateral string")
+			builderCollateral.SetInt64(0)
+		}
+		api.blockBuildersCache[v.BuilderPubkey] = &blockBuilderCacheEntry{
```

Comment:
using big.Int `SetString` would be easier to understand

```suggestion
		builderCollateral, ok := big.NewInt(0).SetString(v.Collateral, 10)
		if !ok {
			api.log.WithError(err).Errorf("could not parse builder collateral string %s", v.Collateral)
		}
		api.blockBuildersCache[v.BuilderPubkey] = &blockBuilderCacheEntry{
```

Reply:
nice! done :-)